src/algorithm.jl

```diff
@@ -557,46 +239,88 @@ The convexification utilized is Tighten McCormick scheme.
             m.status[:bounding_solve] = status
             m.status[:bound] = :Detected
         end
-    elseif status in status_reroute
+        collect_lb_pool(m)    # Always collect details sub-optimal solution
+    elseif status in status_infeasible
         push!(m.logs[:bound], "-")
-        m.status[:bounding_solve] = status
-        print_iis_gurobi(m.model_mip)
-        error("[PROBLEM INFEASIBLE] Infeasibility detected via convex relaxation Infeasibility")
+        m.status[:bounding_solve] = :Infeasible
+        PODDEBUG && print_iis_gurobi(m.model_mip) # Diagnostic code
+        warn("[INFEASIBLE] Infeasibility detected via convex relaxation Infeasibility")
     elseif status == :Unbounded
-        m.status[:bounding_solve] = status
-        error("[MIP UNBOUNDED] MIP solver failure")
+        m.status[:bounding_solve] = :Unbounded
+        warn("[UNBOUNDED] MIP solver return unbounded")
+    else
+        error("[EXCEPTION] MIP solver failure $(status)")
+    end
+
+    return
+end
+
+"""
+    pick_disc_vars(m::PODNonlinearModel)
+
+This function helps pick the variables for discretization. The method chosen depends on user-inputs.
+In case when `indices::Int` is provided, the method is chosen as built-in method. Currently,
+there exist two built-in method:
+
+    * `max-cover(m.disc_var_pick=0, default)`: pick all variables involved in the non-linear term for discretization
+    * `min-vertex-cover(m.disc_var_pick=1)`: pick a minimum vertex cover for variables involved in non-linear terms so that each non-linear term is at least convexified
+
+For advance usage, `m.disc_var_pick` allows `::Function` inputs. User is required to perform flexible methods in choosing the non-linear variable.
+For more information, read more details at [Hacking Solver](@ref).
+
+"""
+function pick_disc_vars(m::PODNonlinearModel)
+
+    if isa(m.disc_var_pick, Function)
+        eval(m.disc_var_pick)(m)
+        length(m.disc_vars) == 0 && length(m.nonconvex_terms) > 0 && error("[USER FUNCTION] must select at least one variable to perform discretization for convexificiation purpose")
+    elseif isa(m.disc_var_pick, Int)
+        if m.disc_var_pick == 0
+            ncvar_collect_nodes(m)
+        elseif m.disc_var_pick == 1
+            min_vertex_cover(m)
+        elseif m.disc_var_pick == 2
+            (length(m.candidate_disc_vars) > 15) ? min_vertex_cover(m) : ncvar_collect_nodes(m)
+        elseif m.disc_var_pick == 3 # Initial
+            (length(m.candidate_disc_vars) > 15) ? min_vertex_cover(m) : ncvar_collect_nodes(m)
+        else
+            error("Unsupported default indicator for picking variables for discretization")
+        end
     else
-        error("[MIP UNEXPECTED] MIP solver failure $(status)")
+        error("Input for parameter :disc_var_pick is illegal. Should be either a Int for default methods indexes or functional inputs.")
```

Comment:
I dont understand the meaning of this error message

Reply:
This means the input for this parameter only takes an integer or a function.